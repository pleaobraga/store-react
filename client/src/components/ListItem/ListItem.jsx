import React from 'react'
import PropTypes from 'prop-types'

import './ListItem.scss'

export const ListItem = ({
  id,
  selected,
  disabled,
  details,
  className,
  actions,
  onChange
}) => {
  return (
    <div className={`list-item ${className ? className : ''}`}>
      <input
        id={id}
        value={selected}
        type="checkbox"
        disabled={disabled}
        onChange={onChange}
      />
      <div className="list-item__details">{details}</div>
      {actions && <div className="list-item__actions">{actions}</div>}
    </div>
  )
}

ListItem.defaultProps = {
  disabled: false,
  selected: false,
  className: '',
  details: {},
  actions: null
}

ListItem.propTypes = {
  id: PropTypes.string,
  onChange: PropTypes.func,
  disabled: PropTypes.bool,
  selected: PropTypes.bool,
  className: PropTypes.string,
  details: PropTypes.node,
  actions: PropTypes.node
}

export default React.memo(ListItem)
