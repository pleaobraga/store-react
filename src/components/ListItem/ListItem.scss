@import '../../styles/colors';

.list-item {
  display: flex;
  width: 100%;
  align-items: center;

  &:hover {
    background-color: $light-gray;

    .list-item__actions {
      visibility: visible;
    }
  }

  &__details {
    display: flex;
    width: 100%;

    * {
      color: $black;

      padding-left: 1rem;
      padding-right: 1rem;

      font-size: 1.8rem;
      font-weight: 400;

      border-top: 1px solid $gray;
      border-bottom: 1px solid $gray;
    }
  }

  &__actions {
    visibility: hidden;

    display: flex;

    i {
      padding-right: 2px;
      cursor: pointer;

      &:hover {
        color: $light-blue;
      }
    }
  }
}
