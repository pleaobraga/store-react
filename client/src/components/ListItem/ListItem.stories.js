import React from 'react'
import { storiesOf } from '@storybook/react'
import { text, number } from '@storybook/addon-knobs'
import { Product } from './ListItem'

export default storiesOf('Components | Product', module).add(
  'default',
  () => (
    <Product
      name={text('Name', 'Name')}
      quantity={number('Quantity', 2)}
      price={text('Price', '20,10')}
      currency={text('Currency', 'R$')}
    />
  ),
  { info: { inline: true, header: false } }
)
