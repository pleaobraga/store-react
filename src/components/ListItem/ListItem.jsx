import React from 'react'
import PropTypes from 'prop-types'

import './ListItem.scss'

export const ListItem = ({ details, className }) => {
  return (
    <div className={`list-item ${className ? className : ''}`}>
      <div className="list-item__details">{details}</div>
      {/* {actions && <div className="list-item__actions">{actions}</div>} */}
    </div>
  )
}

ListItem.defaultProps = {
  className: '',
  details: {},
  actions: {}
}

ListItem.propTypes = {
  className: PropTypes.string,
  details: PropTypes.node,
  actions: PropTypes.node
}

export default React.memo(ListItem)
