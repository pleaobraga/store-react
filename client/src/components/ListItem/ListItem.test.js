import React from 'react'
import { shallow } from 'enzyme'
import ListItem from './ListItem'

describe('ListItem', () => {
  const listItem = shallow(
    <ListItem
      id="test"
      onChange={() => {}}
      disabled={false}
      selected={false}
      className="test"
      details={<div>details</div>}
      actions={<div>actions</div>}
    />
  )

  it('render proprely', () => {
    expect(listItem).toMatchSnapshot()
  })
})
