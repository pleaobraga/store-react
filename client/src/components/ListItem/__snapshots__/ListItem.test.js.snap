// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem render proprely 1`] = `
<div
  className="list-item test"
>
  <input
    disabled={false}
    id="test"
    onChange={[Function]}
    type="checkbox"
    value={false}
  />
  <div
    className="list-item__details"
  >
    <div>
      details
    </div>
  </div>
  <div
    className="list-item__actions"
  >
    <div>
      actions
    </div>
  </div>
</div>
`;
